packages/core/src/libraries/component.ts

```diff
@@ -0,0 +1,197 @@
+import path from "path";
+import * as t from 'io-ts'
+import { ComponentInfo, ComponentMetadata, Preview } from "../models/Library";
+
+import { pascalize } from "../utils/str";
+
+import { getPathToScreenshot } from "./screenshot";
+import Files from "../utils/files";
+import { sliceMocks } from "../mocks";
+import { getOrElseW } from "fp-ts/lib/Either";
+import { Slice, SliceAsObject } from '../models/Slice'
+import { VariationAsObject, AsObject } from '../models/Variation'
+
+import Errors from '../utils/errors'
+
+function getMeta(model: SliceAsObject): ComponentMetadata {
+  return {
+    id: model.id,
+    name: model.name,
+    description: model.description,
+  };
+}
+
+/** take a path to slice and return its name  */
+/* TODO: REFACTOR! so error prone to pop everywhere */
+function getComponentName(slicePath: string): string | undefined {
+  const split = slicePath.split(path.sep);
+  const pop = split.pop();
+  if (!pop) return;
+
+  if (pop.indexOf("index.") === 0) {
+    return split.pop();
+  }
+  if (pop.indexOf(split[split.length - 1]) === 0) {
+    return slicePath.split(path.sep).pop();
+  }
+  return pop.split(".")[0];
+}
+
+/** naive method to find component file in a folder */
```

Comment:
The method is naive because it does not rely on user's possibly custom resolution. This means that a user could define that `.foo` is properly loaded by webpack or equivalent. We cannot know that so it's best effort